support.js

```diff
@@ -58,3 +58,43 @@ var getTemperature = function(zipCode, callback) {
     callback(temp);
   }, waitTime);
 }
+
+
+// The somewhat inaccurate implementation of getTemperatures()
+var getTemperatures = function(zipCodes, callback) {
+  var results = new Object();
+  var length = zipCodes.length;
+  
+  getTemperature(zipCodes[0], function(temp) {
+         results[zipCodes[0].toString()] = temp;
+  });
+  getTemperature(zipCodes[1], function(temp) {
+         results[zipCodes[1].toString()] = temp;
+  });
+  getTemperature(zipCodes[2], function(temp) {
+         results[zipCodes[2].toString()] = temp;
+  });
+  var waitTime = 1000 * 3;
+  _.delay(function() {
+    callback(results);
+  }, waitTime);
+}
+
+/* I try to implement getTemperatures() in a more general way
+var getTemperatures = function(zipCodes, callback) {
+  var results = new Object();
+  var results = [];
+  var len = zipCodes.length;
+  for (index = 0; index < zipCodes.length; index++) {
+      var zip = zipCodes[index];
+      var temp = 0;
+      getTemperature(zip, function(temp) {
+         results[zip.toString()] = temp;
+      });
+  }
+  var waitTime = 1000 * len;
+  _.delay(function() {
```

Comment:
This is a good start.  One challenging thing about this problem is you do not know how long each call to `getTemperature` will take.  Instead of delaying here at the end of the problem, you can put the callback in the `for` loop above once you have received all of the temperatures.